agent/prot/update.go

```diff
@@ -110,13 +111,22 @@ func UpdatePSM(meDID, msgMe string, task *comm.Task, opl didcomm.Payload, subs p
 	} else { // create a new one
 		ss := make([]psm.State, 1, 12)
 		ss[0] = s
+
 		initiator := false
-		if subs&(psm.Sending|psm.Failure) != 0 {
-			glog.V(3).Infof("----- We (%s) are INITIATOR ----", meDID)
-			initiator = true
-		} else {
-			glog.V(3).Infof("----- We (%s) are ADDRESSEE ----", meDID)
+		if task.Role() != "" { // let task define role when it is set
+			initiator = task.Role() == pb.Protocol_INITIATOR.String()
```

Comment:
`pb.Protocol_INITIATOR.String()` should we just use the literal like below, or am I missing something.

Reply:
Just being very sure that there is no typos and comparing the value to the enum that it was filled with 🙂